packages/lib/src/vis/shared/Html.tsx

```diff
@@ -1,145 +1,38 @@
-import { isDefined } from '@h5web/shared';
-import type { GroupProps } from '@react-three/fiber';
-import { useFrame, useThree } from '@react-three/fiber';
-import {
-  useState,
-  useLayoutEffect,
-  useRef,
-  useCallback,
-  useEffect,
-} from 'react';
-import type { HTMLAttributes } from 'react';
+import { useThree } from '@react-three/fiber';
+import { useState, useLayoutEffect } from 'react';
+import type { ReactNode } from 'react';
 import ReactDOM from 'react-dom';
-import type { Group } from 'three';
-import { Vector3 } from 'three';
 
-import { useAxisSystemContext } from './AxisSystemContext';
-
-interface Props extends HTMLAttributes<HTMLDivElement> {
-  groupProps?: GroupProps;
-  followCamera?: boolean;
-  scaleOnZoom?: boolean;
+interface Props {
+  container?: HTMLElement;
+  children?: ReactNode;
 }
 
-// Customised version of drei's `<Html>` component
-// https://github.com/pmndrs/drei/blob/v6.0.3/src/web/Html.tsx
 function Html(props: Props) {
-  const {
-    className,
-    style,
-    children,
-    followCamera,
-    scaleOnZoom,
-    groupProps = {},
-    ...divProps
-  } = props;
-
-  const { width, height } = useThree((state) => state.size);
-  const camera = useThree((state) => state.camera);
   const gl = useThree((state) => state.gl);
-  const { parentElement } = gl.domElement;
-  const { worldToHtml } = useAxisSystemContext();
-
-  // Container `div` for ReactDOM to render into, appended next to R3F's `canvas`
-  const [el] = useState(() => {
-    const elem = document.createElement('div');
-
-    // Take out of flow and let pointer events pass through to canvas
-    elem.style.cssText = `position: absolute; top: 0; left: 0; pointer-events: none;`;
-
-    return elem;
-  });
-
-  // Inner `div` that is transformed and scaled if `followCamera` and `scaleOnZoom` are enabled
-  const innerRef = useRef<HTMLDivElement>(null);
-
-  /* Placeholder R3F group, the position of which is tracked and forwarded to
-   * the inner `div` if `followCamera` and `scaleOnZoom` are enabled. */
-  const group = useRef<Group>(null!); // eslint-disable-line @typescript-eslint/no-non-null-assertion
-
-  const getGroupPosition = useCallback(() => {
-    const worldPos = new Vector3().setFromMatrixPosition(
-      group.current.matrixWorld
-    );
-    const htmlPos = worldToHtml(worldPos);
-
-    return [htmlPos.x, htmlPos.y];
-  }, [worldToHtml]);
+  const { container = gl.domElement.parentElement, children } = props;
```

Comment:
I recommend looking at the diff side by side. So now `Html` can be rendered outside of `VisCanvas`, which opens the door to some interesting refactorings in `AxisSystem`.